File: sistema-escola/src/models/Professor.js
import Pessoa from "./Pessoa.js";

/**
 * Classe Professor
 * Herda de Pessoa e implementa a interface Autenticavel.
 * Atributos privados: #disciplina, #titulacao
 */
class Professor extends Pessoa {
  #disciplina;
  #titulacao;

  constructor(nome, cpf, disciplina, titulacao) {
    super(nome, cpf);
    this.#disciplina = disciplina;
    this.#titulacao = titulacao;
  }

  getDisciplina() { return this.#disciplina; }
  setDisciplina(disciplina) { this.#disciplina = disciplina; }

  getTitulacao() { return this.#titulacao; }
  setTitulacao(titulacao) { this.#titulacao = titulacao; }

  // Implementação da interface Autenticavel
  login() {
    return `Professor ${this.getNome()} acessou o sistema.\n${this.getNome()} está dando aula.`;
  }

  // Implementação do método abstrato de Pessoa
  mostrarDados() {
    return (
      `Professor\n` +
      `Nome: ${this.getNome()}\n` +
      `Disciplina: ${this.#disciplina}\n` +
      `Titulação: ${this.#titulacao}`
    );
  }
}

export default Professor;

File: src/main/java/escola/Professor.java
package escola;

/**
 * Classe Professor
 * Herda de Pessoa e implementa a interface Autenticavel.
 */
public class Professor extends Pessoa implements Autenticavel {

    private String disciplina;
    private String titulacao;

    public Professor(String nome, String cpf, String disciplina, String titulacao) {
        super(nome, cpf);
        this.disciplina = disciplina;
        this.titulacao  = titulacao;
    }

    // Getters e Setters
    public String getDisciplina() { return disciplina; }
    public void   setDisciplina(String disciplina) { this.disciplina = disciplina; }

    public String getTitulacao() { return titulacao; }
    public void   setTitulacao(String titulacao) { this.titulacao = titulacao; }

    // Implementação da interface Autenticavel
    @Override
    public void login() {
        System.out.println("Professor " + getNome() + " acessou o sistema.");
        System.out.println(getNome() + " está dando aula.");
    }

    // Implementação do método abstrato de Pessoa
    @Override
    public void mostrarDados() {
        System.out.println("=== Professor ===");
        System.out.println("Nome:       " + getNome());
        System.out.println("Disciplina: " + disciplina);
        System.out.println("Titulação:  " + titulacao);
    }
}
